### ISSUE TITLE
0.20.4 broke test in gsutil that expected `bar` in `bar@host` to be a password

### ISSUE THREAD

**Issue by NickGoog** [httplib2/httplib2#218]
Possibly related to https://github.com/httplib2/httplib2/pull/217
```
$ pip install httplib2==0.20.2
...
$ gsutil test gslib.tests.test_util.TestUtil.testProxyInfoFromEnvironmentVar
1/1 finished - E[0] F[0] s[0] - TestUtil.testProxyInfoFromEnvironmentVar - .
----------------------------------------------------------------------
Ran 1 test in 0.003s

OK

$ pip install httplib2 --upgrade
...
Successfully installed httplib2-0.20.4

$ gsutil test gslib.tests.test_util.TestUtil.testProxyInfoFromEnvironmentVar
1/1 finished - E[0] F[0] s[0] - TestUtil.testProxyInfoFromEnvironmentVar - F
======================================================================
FAIL: testProxyInfoFromEnvironmentVar (gslib.tests.test_util.TestUtil)
...
AssertionError: None != 'bar'
```

Test code (simplified):
```
def testProxyInfoFromEnvironmentVar(self):
  self.assertEqual(
      httplib2.proxy_info_from_url(
          'http://bar@1.2.3.4:50', method='http').proxy_pass,
      httplib2.ProxyInfo(
          httplib2.socks.PROXY_TYPE_HTTP, '1.2.3.4', 50, proxy_pass='bar').proxy_pass)
```

Notes:
* Testing against 0.20.2 b/c I didn't see a 0.20.3 under PIP releases.
* Heavily simplified the actual gustil test case, which is why the test body has nothing to do with the title. The discrepency between versions 0.20.2 and 0.20.4 is reproducible with this short piece of code.

Thanks for any help!

**Comment by temoto** [comment]
@NickGoog `bar` is obviously not a password, because there is no username and colon before it, like in `user:pass`. Proper test should've been failing with previous httplib2 code, but it's correct now. We're using python stdlib `urlparse` since #217 released in 0.20.4.

I'm sorry it broke external tests, I expected backward compatibility, would've at least increase minor version and write big warning in changelog.

Filed fix in gsutil test.

Have an awesome day.

**Comment by NickGoog** [comment]
No problem, I also thought the fact it was getting interpreted as "password" and not "user" was a bit strange.
Thanks for clarifying!
